source/tests/pt/test_dp_show.py

```diff
@@ -0,0 +1,288 @@
+# SPDX-License-Identifier: LGPL-3.0-or-later
+import json
+import os
+import shutil
+import unittest
+from copy import (
+    deepcopy,
+)
+from pathlib import (
+    Path,
+)
+from typing import (
+    ClassVar,
+)
+
+import torch
+
+from deepmd.pt.entrypoints.main import (
+    get_trainer,
+)
+from deepmd.pt.utils import (
+    env,
+)
+from deepmd.pt.utils.multi_task import (
+    preprocess_shared_params,
+)
+
+from .model.test_permutation import (
+    model_se_e2_a,
+)
+
+
+# mock FLAGS object
+class FLAGS:
+    INPUT: ClassVar[str] = ""
+    ATTRIBUTES: ClassVar[list] = []
+
+
+# mock log
+class MockLogger:
+    def __init__(self):
+        self.messages = []
+
+    def info(self, message):
+        self.messages.append(message)
+
+
+# mock show function, the only difference from origin function is
+# add `log = MockLogger()` and `return log.messages`
+def mock_show(FLAGS):
+    log = MockLogger()
```

Comment:
Could you use the function in `deepmd/pt/entrypoints/main.py` instead of manually coping it here? 
UT means that if you changed the **original function** in `deepmd/pt/entrypoints/main.py`, this UT must reflect the change.

To get the screen output, you may use `output = os.system(...)` to achieve that.

Reply:
I try some methods to get the screen output, but they fail.
1. 
`import subprocess`
 `command = "dp --pt show multi.pt model-branch"`
`result = subprocess.check_output(command, shell=True, text=True)`
`print("Result:",result)`
output: `Result:           `
2.
`import os`
`command = "dp --pt show multi.pt model-branch"`
`result = os.popen(command).read()`
`print("Output from command:", result)`
output: `Output from command:         `
3. 
`import os`
`command = "dp --pt show multi.pt model-branch"`
`output_file = "output.txt"`
`execute = os.system(f"{command} > {output_file}")`
`if execute == 0:`
`    print("Successfully executed")`
`else:                   `
`    print("Failed")`
but the `output.txt` is empty.

Reply:
Do you have any experience with this?

Reply:
logging prints to stderr instead of stdout

Reply:
It works. I will modify the UT.